### ISSUE TITLE
HyperHTMLElement.define('my-tag', class {...}) please

### ISSUE THREAD

**Issue by jfbrennan** [WebReflection/hyperHTML-Element#40]
Was hoping to be able to do
```
HyperHTMLElement.define('my-component', class {
  created() {}
  ...
});
```
instead of 
```
class MyComponent extends HyperHTMLElement {
  created() {}
  ...
}

HyperHTMLElement.define('my-component');
```
Doesn't seem to work. Can this be made possible?

Btw, really happy to find another solution that's staying close to standards. I think libraries like this and Riot.js and others are the future and will have a longer life because of the decision to stay close to standards.

**Comment by WebReflection** [comment]
> Can this be made possible?

If you don't extend `HyperHTMLElement` you won't have its goodies, unless it overwrites patching all the things which is not desirable.

The `define` in HyperHTMLElement does more than just defining, and inheritance matters with Custom Elements.

TL;DR it's not really possible, unless you'd go for `HyperHTMLElement.define('my-component', class extends HyperHTMLElement { ... });` which is ugly as hell.

Remember: `define` returns the class itself, so if it's about having classes expressions, you can always do:

```js
const exp = (class extends HyperHTMLElement {}).defiune('my-tag');
```

**Comment by jfbrennan** [comment]
I corrected a mistake in my code sample, but doesn't matter. Thanks for answering the question
